### PR TITLE
2018-day-2

### PR DESCRIPTION
**Pull request by Herrera93**
- Created library to get checksum and common letters from a series of strings.
- To solve day 2 2018 part 1 & 2

## PR - Merge Checklist
- [x] CR'd
- [x] Workflow: Trello card is in "Code Review" column
- [x] README updated or N/A
- [x] Benchmarks run and compared with master or N/A

### CODE REVIEW

**Comment by Herrera93** [comment]
[2018-day-2-part-1](https://trello.com/c/2wBIuN9R/58-2018-day-2-part-1)

**Comment by Herrera93** [comment]
[2018-day-2-part-2](https://trello.com/c/xxDBOC4D/59-2018-day-2-part-2)

File: src/year2018/day_2.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+use std::collections::HashMap;
+
+pub fn get_checksum(ids: &[String]) -> i32 {
+    // Find the number of strings that have exactly three letters and
+    // the number of string that have exactly two letters. The multiply
```
Comment (kanekv) [inline src/year2018/day_2.rs]: `string` -> `strings`
`The` -> `Then`

File: src/year2018/day_2.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    // them to get the checksum
+    let (twos, threes) = ids
+        .iter()
+        .map(|s| {
+            s.chars()
+                .fold(HashMap::new(), |acc: HashMap<char, i32>, c| {
+                    let mut r = acc.clone();
```
Comment (kanekv) [inline src/year2018/day_2.rs]: Why are you cloning accelerator? Isn't there a way to create a mutable one?

File: src/year2018/day_2.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    2 => (1, acc.1),
+                    3 => (acc.0, 1),
+                    _ => acc,
+                })
+        }).fold((0, 0), |acc, x| (acc.0 + x.0, acc.1 + x.1));
+    twos * threes
+}
+
+pub fn get_common_letters(ids: &[String]) -> String {
+    // Find the common letters between two string that have exatcly
```
Comment (kanekv) [inline src/year2018/day_2.rs]: `exatcly` -> `exactly`

File: src/year2018/day_2.rs (hunk 2 above)
Comment (Herrera93) [reply src/year2018/day_2.rs]: You're right, let me change that.

**Pull request merged**